### PR TITLE
Feature/integration tests

### PR DESCRIPTION
**Pull request by tanvirchahal**
In reference to https://github.com/datopian/data-explorer-graphql/issues/45 

- Added tests 
- Added github action to test on each push

Dataset it expects to test: transmissionlines

Note: currently the tests are failing in github CI, needs more work

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,5 +25,11 @@ jobs:
       - name: Install dependencies
         run: yarn
 
-      - name: Run the tests
+      - name: Run the unit tests
         run: yarn test
+
+      - name: Cypress run
```
Comment (anuveyatsu) [inline .github/workflows/test.yml:31]: @tanvirchahal let's call it end to end tests or integration tests so it's more accurate

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,5 +25,11 @@ jobs:
...
+        uses: cypress-io/github-action@v2
+        with:
+          start: yarn start
```
Comment (anuveyatsu) [inline .github/workflows/test.yml:34]: @tanvirchahal I expected `yarn e2e` command here. Why we have `yarn start` which is for starting the server?

File: package.json [hunk 3]
```diff
@@ -22,6 +22,9 @@
     "start": "craco start",
     "build": "craco build",
     "test": "craco test",
+    "cypress:open": "cypress open",
+    "cypress:ci": "cypress run --config video=false",
```
Comment (anuveyatsu) [inline package.json:26]: @tanvirchahal I think this is the command you need to run in CI so it passes

Comment (anuveyatsu) [reply package.json:26]: it would be great if we could rename this command to `e2e:ci`

File: package.json [hunk 4]
```diff
@@ -22,6 +22,9 @@
     "start": "craco start",
     "build": "craco build",
     "test": "craco test",
+    "cypress:open": "cypress open",
```
Comment (anuveyatsu) [inline package.json:25]: and this one can be `e2e:interactive`

File: public/index.html [hunk 5]
```diff
@@ -9,7 +9,7 @@
     <noscript>You need to enable JavaScript to run this app.</noscript>
     <div
       id="root"
-      data-api="http://localhost:3333/v1/"
+      data-api="https://data-api.energidataservice.dk/v1/"
```
Comment (anuveyatsu) [inline public/index.html:12]: @tanvirchahal this shouldn't be hardcoded - we need to find solution to make configurable

File: src/Components/FilterComponents/OrderBy.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ function OrderBy({ orderColumnRef, orderByRef, fields }) {
         ref={orderColumnRef}
         className="mr-2 border"
         data-testid="data-ord"
+        id="data-ord"
```
Comment (anuveyatsu) [inline src/Components/FilterComponents/OrderBy.js:10]: @tanvirchahal why it is needed?

File: src/Components/FilterComponents/SelectFilter.js [hunk 7]
```diff
@@ -98,7 +98,7 @@ function SelectFilter({
   }
 
   return (
-    <div className="mb-2" data-testid="field-container">
+    <div className="mb-2" data-testid="field-container" id="field-container">
```
Comment (anuveyatsu) [inline src/Components/FilterComponents/SelectFilter.js:101]: @tanvirchahal issue here is that `id` must be unique but you may have several Select Filters I guess

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
@tanvirchahal left some comments, please, have a look

File: .github/workflows/test.yml (hunk 1 above)
Comment (tanvirchahal) [reply .github/workflows/test.yml]: @anuveyatsu changed the name to "Run the integration tests"

File: .github/workflows/test.yml (hunk 2 above)
Comment (tanvirchahal) [reply .github/workflows/test.yml]: @anuveyatsu i tried running tests without `yarn start` and i get the following error 

`Cypress could not verify that this server is running:
 > http://localhost:3000/`
 
 Full error log: https://github.com/datopian/data-explorer-graphql/runs/2677912757?check_suite_focus=true
 
 I think it's important to run the server before cypress can get started. 
 
 I got the yarn start code from official documentation for cypress: https://docs.cypress.io/guides/continuous-integration/github-actions#Basic-Setup

File: package.json (hunk 3 above)
Comment (tanvirchahal) [reply package.json]: @anuveyatsu Updated the commands to be 
`"e2e:interactive": "cypress open",
    "e2e:ci": "cypress run --config video=false",
    "e2e": "cypress run"`

File: package.json (hunk 4 above)
Comment (tanvirchahal) [reply package.json]: Updated the command

File: src/Components/FilterComponents/SelectFilter.js (hunk 7 above)
Comment (tanvirchahal) [reply src/Components/FilterComponents/SelectFilter.js]: @anuveyatsu refactored code to use `data-testid` and removed all new added IDs

File: src/Components/FilterComponents/OrderBy.js (hunk 6 above)
Comment (tanvirchahal) [reply src/Components/FilterComponents/OrderBy.js]: @anuveyatsu refactored code to use existing `data-testid` and removed all new added IDs

File: public/index.html (hunk 5 above)
Comment (tanvirchahal) [reply public/index.html:12]: @anuveyatsu how important is it to put the `data-api` in `public/index.html`? 

I suggest that we can put these variables in a `.env` file and load them directly into `Index.js` component  

Then we can have two files `.env` and `.env.test` which can have separate `data-api` URLs 

If its important to keep the `data-api` in `public/index` , then we can try loading these variables from an external JSON file before loading the application but I am not sure how stable would that be.

File: .github/workflows/test.yml (hunk 2 above)
Comment (anuveyatsu) [reply .github/workflows/test.yml:34]: OK got it now 👍

File: public/index.html (hunk 5 above)
Comment (anuveyatsu) [reply public/index.html:12]: @tanvirchahal ok, I think we can keep it for now 👍

**Review by anuveyatsu — APPROVED** [review]
LGTM

**Pull request merged**